stdlib/Printf/src/Printf.jl

```diff
@@ -262,11 +262,17 @@ end
 end
 
 # integers
+toint(x) = x
+toint(x::Rational) = Integer(x)
```

Comment:
Doesn't this need `round` too?

Reply:
I was just matching old printf behavior here:
```julia
julia> @sprintf("%d", 3//2)
ERROR: InexactError: Integer(3//2)
Stacktrace:
 [1] Integer at ./rational.jl:109 [inlined]
 [2] decode_dec at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Printf/src/Printf.jl:855 [inlined]
 [3] decode_dec(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Rational{Int64}, ::String, ::Int64, ::Int64, ::Char, ::Array{UInt8,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Printf/src/Printf.jl:841
 [4] top-level scope at REPL[4]:1

julia> @sprintf("%d", 3//1)
"3"
```
But I could see the case that we should call `round` first

Reply:
Right, it just seemed like the direction we have taken is to integerify any number (admittedly a bit weird -- would be simpler to force the user to do it for `%d`). Perhaps the fallback should just be `round(Int, x)` instead of a no-op?